compiler+runtime/src/cpp/jank/runtime/module/loader.cpp

```diff
@@ -477,15 +457,30 @@ namespace jank::runtime::module
       return res;
     }
 
-    loader::set_loaded(module);
-
     return ok();
   }
 
   string_result<void>
   loader::load_o(native_persistent_string const &module, file_entry const &entry) const
   {
     profile::timer const timer{ fmt::format("load object {}", module) };
+
+    /* While loading an object, if the main ns loading symbol exists, then
+     * we don't need to load the object file again.
+     *
+     * Existence of the `jank_load_<module>` symbol (also a function),
+     * means that all the required symbols exist and are already defined.
+     * We call this symbol to re-initialize all the vars in the namespace.
+     * */
```

Comment:
So this means that it'll be impossible, in jank, to:

1. Load a module from an object file
2. Change the source behind that module
3. Try to require the module again, to reload it
4. Get the updated source

What does Clojure do here?

Reply:
If the source is more recent than the object file, we will always load from source.

Clojure doesn't load from `class` files, while `reloading` in the repl.

Reply:
Let me know, if this makes sense?